src/solve.jl

```diff
@@ -261,6 +269,16 @@ function DiffEqBase.__init(
     # automatically determine initial time step
     if iszero(dde_int.dt) && dde_int.opts.adaptive
         auto_dt_reset!(dde_int)
+        if sign(dde_int.dt)!=dde_int.tdir && dde_int.dt!=tType(0) && !isnan(dde_int.dt)
+            error("Automatic dt setting has the wrong sign. Exiting. Please report this error.")
+        end
+        if isnan(dde_int.dt)
+            if verbose
+                @warn("Automatic dt set the starting dt as NaN, causing instability.")
+            end
+        end
+    elseif dde_int.opts.adaptive && dde_int.dt > zero(dde_int.dt) && dde_int.tdir < 0
+        dde_int.dt *= dde_int.tdir # Allow positive dt, but auto-convert
```

Comment:
This is just a copy of https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/solve.jl#L328-L340, so it could be simplified e.g. by defining this code block with `@def` in OrdinaryDiffEq.

Reply:
We could make this a function.